core/org.wso2.carbon.core/src/main/java/org/wso2/carbon/core/util/KeyStoreManager.java

```diff
@@ -437,6 +495,67 @@ public PrivateKey getDefaultPrivateKey() throws Exception {
                 "available only for the super tenant.");
     }
 
+    /**
+     * This method loads the private key of a given tenant key store
+     *
+     * @param keyStoreName name of the tenant key store
+     * @param alias        alias of the private key
+     * @return private key corresponding to the alias
+     */
+    public PrivateKey getTenantPrivateKey(String keyStoreName, String alias) throws Exception{
```

Comment:
Do we expect to call this method outside this class?

Reply:
Fixed in https://github.com/wso2/carbon-kernel/pull/4025/commits/9ee2fb4ee41a2ba18b38dc41bd028c113f15ee28